unit_test/solver/matrix/test_matrix.py

```diff
@@ -0,0 +1,206 @@
+
+        # act
+        matrix.add_equation(equation_object=equation)
+
+        # assert
+        self.assertEqual(len(matrix.mat), 1)
+        self.assertEqual(len(matrix.rhs), 1)
+        self.assertEqual(matrix.mat[0], [1.0, 1.0])
+        self.assertEqual(matrix.rhs[0], 10.0)
+
+    def test_solve(self):
+        """Test the solving of the matrix object."""
+        # arrange
+        matrix = Matrix()
+        index = matrix.add_unknowns(2)
+        equation1 = EquationObject()
+        equation1.indices = np.array([index, index + 1])
+        equation1.coefficients = np.array([1.0, 1.0])
+        equation1.rhs = 0.0
+        equation2 = EquationObject()
+        equation2.indices = np.array([index, index + 1])
+        equation2.coefficients = np.array([0.0, 1.0])
+        equation2.rhs = 10.0
+
+        # act
+        result = matrix.solve([equation1, equation2])
+
+        # assert
+        self.assertEqual(result, [-10.0, 10.0])
+
+    def test_solve_large(self):
+        """Test the solving of the matrix object."""
+        # arrange
+        matrix = Matrix()
+        value = 10.0
+        equations = []
+        size = 1000
+        for _ in range(size):
+            position = matrix.add_unknowns(1)
+            equation_object = EquationObject()
+            equation_object.indices = [0]
+            equation_object.rhs = value
+            equation_object.indices = [position]
+            equation_object.coefficients = [1.0]
+            equations.append(equation_object)
+
+        # act
+        results = matrix.solve(equations)
+
+        # assert
+        self.assertEqual(results, [value] * size)
+
+    def test_is_converged(self):
+        """Test the is converged of the matrix object."""
+        # arrange
+        matrix = Matrix()
+        matrix.add_unknowns(2)
+
+        # act
+        result = matrix.is_converged()
+
+        # assert
+        self.assertEqual(result, False)
+
+    def test_convergence_raise_error(self):
+        """Test the convergence of the matrix object."""
+        # arrange
+        matrix = Matrix()
+
+        # act
+        matrix.relative_convergence = -1
+
+        # assert
+        with self.assertRaises(ValueError):
+            matrix.is_converged()
+
+    def test_is_converged2(self):
+        """Test the is converged of the matrix object."""
+        # arrange
+        matrix = Matrix()
+        matrix.add_unknowns(1)
+        equation = EquationObject()
+        equation.indices = [0]
+        equation.coefficients = [1.0]
+        equation.rhs = 10.0
+        matrix.solve([equation])
+        matrix.solve([equation])
+
+        # act
+        result = matrix.is_converged()
+
+        # assert
+        self.assertEqual(result, True)
+
+    def test_get_solution(self):
+        """Test the get solution of the matrix object."""
+        # arrange
+        matrix = Matrix()
+        matrix.add_unknowns(10)
+
+        # act
+        result = matrix.get_solution(0, 2)
```

Comment:
Voeg hier ook even de naam toe. 

Ik zou ook de sol_new vervangen door specifieke waardes. Je checkt nu niet of je de juiste index pakt want hij bevat allemaal ones!

Reply:
Aangepast